File: 1661-minimum-number-of-vertices-to-reach-all-nodes/1661-minimum-number-of-vertices-to-reach-all-nodes.java
class Solution {
    public List<Integer> findSmallestSetOfVertices(int n, List<List<Integer>> edges) {
        int[] inDegree = new int[n];
        for (List<Integer> list : edges) {
            inDegree[list.get(1)]++;
        }
        List<Integer> res = new ArrayList<>();
        for (int i = 0; i < n; i++) {
            if (inDegree[i] == 0) {
                res.add(i);
            }
        }
        return res;
    }
}

File: 1661-minimum-number-of-vertices-to-reach-all-nodes/1661-minimum-number-of-vertices-to-reach-all-nodes.py
class Solution:
    def findSmallestSetOfVertices(self, n: int, edges: List[List[int]]) -> List[int]:
        in_degree = [0] * n
        for _, b in edges:
            in_degree[b]+=1     
        res = []
        for i in range(n):
            if in_degree[i] == 0:
                res.append(i)
        return res   
